### PR TITLE
Using rust_link in FFI

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Comment by robertbastian** [comment]
Hmm I manually validated all URLs and found a typo and one that was out of date. Maybe the Diplomat tool could check that the URLs are 200s?

**Comment by Manishearth** [comment]
>  Maybe the Diplomat tool could check that the URLs are 200s?

Strongly against a codegen tool hitting the network. As a separate linkcheck tool, maybe, but it's unexpected when such tools hit the network.

**Comment by robertbastian** [comment]
It could be a `HEAD` request behind a flag

**Comment by sffc** [comment]
It looks like you update the `diplomat` version. Look in `ffi/diplomat/Cargo.toml` and change the sha of diplomat.

**Comment by Manishearth** [comment]
Also the rust_link support is broken in diplomat and https://github.com/rust-diplomat/diplomat/pull/138 needs to land first

**Review by sffc — COMMENTED** [review]
(setting review bit)

**Review by Manishearth — APPROVED** [review]

File: ffi/diplomat/diplomat_gen_all.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
-diplomat-tool c ./c/include
-diplomat-tool cpp ./cpp/include
-diplomat-tool js ./wasm/lib/ ./wasm/docs.rst
+diplomat-tool c ./c/include --docs-base-urls=icu:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=fixed_decimal:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=icu_plurals:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=icu_properties:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=icu_codepointtrie:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=icu_uniset:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=icu_provider_blob:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=icu_provider_fs:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=icu_segmenter:https://unicode-org.github.io/icu4x-docs/doc/ --docs-base-urls=icu_provider:https://unicode-org.github.io/icu4x-docs/doc/
```
Comment (Manishearth) [inline ffi/diplomat/diplomat_gen_all.sh:2]: issue: I'd rather not do this, this is brittle

As I'd proposed [here](https://github.com/rust-diplomat/diplomat/pull/136#discussion_r835496570), can we have a single "default" parameter for the doc base URL? we could continue to allow for passing in additional stuff. And we can special case `docs.rs` to mean "link to docs.rs with the correct format"

**Review by Manishearth — CHANGES_REQUESTED** [review]

**Review by Manishearth — APPROVED** [review]

**Pull request merged**